rllib/policy/sample_batch.py

```diff
@@ -303,12 +329,52 @@ def timeslices(self, k: int) -> List["SampleBatch"]:
             List[SampleBatch]: The list of (new) SampleBatches (each one of
                 size k).
         """
-        out = []
-        i = 0
-        while i < self.count:
-            out.append(self.slice(i, i + k))
-            i += k
-        return out
+        slices = self._get_slice_indices(k)
+        timeslices = [self.slice(i, j) for i, j in slices]
+        return timeslices
+
+    def zero_pad(self, max_seq_len: int, exclude_states=True):
+        """Zero pad the data in this SampleBatch in place.
```

Comment:
Probably clean up code here. The logic is a bit messy

Reply:
done